src/runmode-odp.c

```diff
@@ -0,0 +1,172 @@
+/* Copyright (C) 2016 Linaro
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Maxim Uvarov <maxim.uvarov@linaro.org>
+ *
+ * OpenDataPlane runmode support
+ */
+
+#include "suricata-common.h"
+#include "config.h"
+#include "tm-threads.h"
+#include "conf.h"
+#include "runmodes.h"
+#include "runmode-netmap.h"
+#include "log-httplog.h"
+#include "output.h"
+#include "detect-engine-mpm.h"
+
+#include "alert-fastlog.h"
+#include "alert-prelude.h"
+#include "alert-unified2-alert.h"
+#include "alert-debuglog.h"
+
+#include "util-debug.h"
+#include "util-time.h"
+#include "util-cpu.h"
+#include "util-affinity.h"
+#include "util-device.h"
+#include "util-runmodes.h"
+#include "util-ioctl.h"
+
+#ifdef HAVE_ODP
+
+#include "source-odp.h"
+#include <odp_api.h>
+
+/* Maximum ODP packet size */
+#define POOL_SIZE 9000
+/** Number of packets in the pool */
+#define POOL_NUM  1000
+
+odp_instance_t instance;
+extern int max_pending_packets;
+
+static const char *default_mode_workers = NULL;
+
+const char *RunModeODPGetDefaultMode()
+{
+    return default_mode_workers;
+}
+
+static void *ParseODPConfig(const char *iface_name)
+{
+    ODPIfaceConfig *conf = SCMalloc(sizeof(*conf));
+
+    if (unlikely(conf == NULL)) {
+        return NULL;
+    }
+
+    return conf;
+}
+
+static int ODPConfigGeThreadsCount(void *conf)
+{
+    odp_cpumask_t cpumask;
+    int max_cpus = 0; /* all available worker cpus */
+
+    return odp_cpumask_default_worker(&cpumask, max_cpus);
```

Comment:
I I understand correctly we will use maximum of CPUs available on the system as number of threads. I'm not sure of the target use case but this could mean dedicating the box to Suricata. Maybe we should have an YAML option to set the number of threads and be able to share the system with other one.